model_analyzer/result/result_comparator.py

```diff
@@ -0,0 +1,202 @@
+            1
+                if result1 > result2
+            -1
+                if result1 < result2
+        """
+
+        # For now create an average measurment
+        agg_measurement1 = self._aggregate_measurements(
+            result=result1, aggregation_func=average_list)
+        agg_measurement2 = self._aggregate_measurements(
+            result=result2, aggregation_func=average_list)
+
+        return self.compare_measurements(measurement1=agg_measurement1,
+                                         measurement2=agg_measurement2)
+
+    def compare_measurements(self, measurement1, measurement2):
+        """
+        Compares individual meausurements retrieved from perf runs
+        based on priorities specified by model analyzer config
+
+        Parameters
+        ----------
+        measurement1 : Measurement
+            first set of (gpu_measurements, non_gpu_measurements) to
+            be compared
+        measurement2 : Measurement
+            first set of (gpu_measurements, non_gpu_measurements) to
+            be compared
+
+        Returns
+        -------
+        int 
+            0 
+                if the results are determined
+                to be the same within a threshold
+            1
+                if measurement1 > measurement2
+            -1
+                if measurement1 < measurement2
+        """
+
+        gpu_data1 = measurement1.gpu_data()
+        gpu_data2 = measurement2.gpu_data()
+
+        gpu_rows1 = []
+        gpu_rows2 = []
+
+        for _, gpu_row in gpu_data1.items():
+            gpu_rows1.append(gpu_row)
+        for _, gpu_row in gpu_data2.items():
+            gpu_rows2.append(gpu_row)
+
+        avg_gpu_data1 = average_list(gpu_rows1)
+        avg_gpu_data2 = average_list(gpu_rows2)
+
+        non_gpu_data1 = measurement1.non_gpu_data()
+        non_gpu_data2 = measurement2.non_gpu_data()
+
+        for priority in self._metric_priorities:
+            if priority in self._non_gpu_type_to_idx:
+                # Get position in measurements of current priority's value
+                metric_idx = self._non_gpu_type_to_idx[priority]
+                threshold = self._comparison_threshold_factor * non_gpu_data1[
+                    metric_idx].value()
+                value_diff = (non_gpu_data1[metric_idx] -
+                              non_gpu_data2[metric_idx]).value()
+
+                if value_diff > threshold:
+                    return 1
+                elif value_diff < -threshold:
+                    return -1
+            elif priority in self._gpu_type_to_idx:
+                metric_idx = self._gpu_type_to_idx[priority]
+                threshold = self._comparison_threshold_factor * avg_gpu_data1[
+                    metric_idx].value()
+                value_diff = (avg_gpu_data1[metric_idx] -
+                              avg_gpu_data2[metric_idx]).value()
+
+                if value_diff > threshold:
```

Comment:
What about comparing the absolute value of the difference and threshold instead?

Reply:
wouldn't that remove the information about which is greater? `abs(value_diff) < threshold` would allow me to `continue` but within `abs(value_diff) > threshold` I would need to add further checks of the sort I've used here.

Reply:
Right.